### PR TITLE
Update ExperimentStatusIndicator

### PR DESCRIPTION
**Pull request by ahdriel**
### Features and Changes

- Updated `ExperimentStatusIndicator` with new states
  - `Stopped` now includes the result.
  - The component can show either the main status, only the details, or both (the screenshot shows both)
- Updated `ResultsIndicator` to also use the new Badges
  - They are sometimes shown together, depending on what we want to call attention to
 
Screenshot 2024-12-17 at 4 55 39 PM
Screenshot 2024-12-17 at 4 59 54 PM

### More screenshots

|  |  |  |  |  |
|------|------|------|------|------|
| Experiment List | [Before](https://github.com/user-attachments/assets/9c666829-a3f1-4ad7-b444-ff2608655463) | [After](https://github.com/user-attachments/assets/97c57ce8-3c3e-40f7-8dc3-f5625fdaa948) | | |
| Draft | [Before](https://github.com/user-attachments/assets/002aa39b-209f-4569-833f-05a87dc1f550) | [After](https://github.com/user-attachments/assets/fcf845e0-b870-45b6-9b04-28e5b5494fbb) | [Before (Archived)](https://github.com/user-attachments/assets/1ffd9481-9dc8-4b9e-8199-2ffc2f394311) | [After (Archived)](https://github.com/user-attachments/assets/2bd8af52-15b6-48fe-8716-004db90d8134) |
| Running | [Before](https://github.com/user-attachments/assets/140c242e-d084-49ed-902d-e6d347b54ffe) | [After](https://github.com/user-attachments/assets/577b0b69-bacd-4b9f-9067-5610aa2c7e99) | [Before (Archived)](https://github.com/user-attachments/assets/e136fcae-f614-462a-a1b1-4cbaeccb4e5e) | [After (Archived)](https://github.com/user-attachments/assets/6b9169e8-b5dc-4974-b04d-9ff8831d7c4a) |
| Stopped - Did not finish | [Before](https://github.com/user-attachments/assets/bb3a3517-4c2c-43eb-81a4-7594554664a3) | [After](https://github.com/user-attachments/assets/84878e38-3aba-437f-ab09-1b71b7b71944) | [Before (Archived)](https://github.com/user-attachments/assets/c7e81620-f82d-4670-8c32-792b53a5b0bf) | [After (Archived)](https://github.com/user-attachments/assets/6a36d4d4-fca4-42c3-a9e3-14ad54fcae70) |
| Stopped - Inconclusive | [Before](https://github.com/user-attachments/assets/e2a10ed7-a38d-4f61-ae0e-3e1167f188e1) | [After](https://github.com/user-attachments/assets/c582f978-2d7e-44ce-9950-0be98fc690fd) | [Before (Archived)](https://github.com/user-attachments/assets/2172b465-7a5d-4f09-9c72-042fb863ab65) | [After (Archived)](https://github.com/user-attachments/assets/cfb6e29e-7989-423b-aa07-1398769ee623) |
| Stopped - Lost | [Before](https://github.com/user-attachments/assets/0723a908-0705-4b12-821c-d33a6712625c) | [After](https://github.com/user-attachments/assets/627bf734-68ad-42e1-bcd8-399c114b0f0b) | [Before (Archived)](https://github.com/user-attachments/assets/6e9c01a6-dafc-4a36-81cc-e60fc95067e5) | [After (Archived)](https://github.com/user-attachments/assets/0ca45b02-8dc9-48dd-9cba-b15efb1a6a34) |
| Stopped - Waiting | [Before](https://github.com/user-attachments/assets/ab85ea91-80f6-4a78-9eda-8560ff3aae73) | [After](https://github.com/user-attachments/assets/a358f747-2284-432c-8874-d63c7379b50e) | [Before (Archived)](https://github.com/user-attachments/assets/bf1bc8e0-17a1-4e72-9e0b-9ec962d971b5) | [After (Archived)](https://github.com/user-attachments/assets/0bc0b8f8-386f-4999-a4ee-372ffc984dc1) |
| Stopped - Won | [Before](https://github.com/user-attachments/assets/d25b55f3-6f92-4bd6-ac10-4d5882f31583) | [After](https://github.com/user-attachments/assets/89fdf919-2739-46cc-a74b-bdc79b36f53b) | [Before (Archived)](https://github.com/user-attachments/assets/bb8ce895-a7c5-40ca-a6b8-51fc90e34950) | [After (Archived)](https://github.com/user-attachments/assets/e826a506-de40-4f6f-a854-e0bdf9b882f5) |

### CODE REVIEW

**Comment by jdorn** [comment]
This design isn't ready to be worked on yet.  We're planning to split this into 2 parts and add a bunch more variations of the "running" status.
- Top-level status: Either `draft`, `running`, `stopped`, or `archived`
- Sub-status:
  - When status = "stopped": `Won`, `Lost`, Didn't Finish`, `Inconclusive`, `Awaiting Decision`
  - When status = "running": TBD, but things like "Ready to Ship", "Gathering Data", etc.

**Comment by jdorn** [comment]
I think I'd prefer to wait.  There were some concerns with the colors as well (e.g. getting rid of green for "won" and making "Didn't Finish" stand out so much more than the others)

File: packages/front-end/components/Experiment/TabbedPage/ExperimentStatusIndicator.tsx [hunk 1]
```diff
@@ -1,36 +1,116 @@
...
+
+  if (experimentData.status == "running") {
+    return ["indigo", "solid", "Running"];
+
+    // TODO: Add detail statuses
+    // return ["indigo", "solid", "Running", "~5 days left"];
+    // return ["amber", "solid", "Running", "Unhealthy"];
+    // return ["amber", "soft", "Running", "Rollback now"];
+    // return ["amber", "soft", "Running", "Ship now"];
+    // return ["amber", "soft", "Running", "Discuss results"];
```
Comment (mknowlton89) [inline packages/front-end/components/Experiment/TabbedPage/ExperimentStatusIndicator.tsx:63]: Are these detailed statuses available now or do they rely on the decision making framework?

File: packages/front-end/components/MetricExperiments/MetricExperiments.tsx [hunk 2]
```diff
@@ -188,7 +188,10 @@ function MetricExperimentResultTab({
         </td>
         <td>
           <div className="my-1">
-            <ExperimentStatusIndicator status={e.status} />
+            {/* FIXME: How can we get the archived status? */}
```
Comment (mknowlton89) [inline packages/front-end/components/MetricExperiments/MetricExperiments.tsx:191]: Looks like we'll need to update `getExperimentsUsingMetrics` ([here](https://github.com/growthbook/growthbook/blob/4da03b9d283d5c6e2034bad3fc54778f9d6595b4/packages/back-end/src/models/ExperimentModel.ts#L661)) - Probably makes sense to add an optional prop for this that defaults to not including archived experiments.

**Review by mknowlton89 — APPROVED** [review]
Looks good - tested on my end and didn't see anything that needs addressing.

Left a few comments on the comments.

File: packages/front-end/components/Experiment/TabbedPage/ExperimentStatusIndicator.tsx (hunk 1 above)
Comment (ahdriel) [reply packages/front-end/components/Experiment/TabbedPage/ExperimentStatusIndicator.tsx:63]: Yeah they rely on the decision framework -- the first one we plan on shipping is Unhealthy.

So while we don't have it at the moment, wanted to keep this list in code so we remember about it -- but also happy to remove it and reference Figma instead.

Comment (mknowlton89) [reply packages/front-end/components/Experiment/TabbedPage/ExperimentStatusIndicator.tsx:63]: Nah - happy to leave them in for reference, just wanted to confirm my understanding.

**Pull request merged**